lib/charms/opensearch/v0/opensearch_profile.py

```diff
@@ -0,0 +1,274 @@
+        """Get the cluster topology requirements for this profile."""
+        return ClusterTopologyRequirements(
+            cluster_managers=1,
+            data=1,
+        )
+
+
+class ProfilesManager:
+    """Manage all profile related operations"""
+
+    def __init__(self, state: "OpenSearchClusterState", workload: "OpenSearchDistribution"):
+        self.state = state
+        self.workload = workload
+        try:
+            if self.profile.type == PerformanceType.TESTING:
+                logger.warning(
+                    "Testing profile is used. This profile is not suitable for production use and should only be used for testing purposes."
+                )
+        except ValueError:
+            logger.error(
+                "Invalid profile configuration. Value: %s", self.state.config.get("profile")
+            )
+
+    def check_missing_system_requirements(self) -> List[str]:
+        """Checks the system requirements."""
+        return self.workload.check_missing_system_requirements()
+
+    def check_memory_requirements(self, profile: OpenSearchProfile) -> List[str]:
+        """Checks memory requirements for the unit."""
+        memory_size = self.workload.meminfo()["MemTotal"]
+
+        if (
+            profile.memory_requirements.memory_size
+            and memory_size < profile.memory_requirements.memory_size
+        ):
+            logger.error(
+                "Insufficient memory: %s < %s",
+                memory_size,
+                profile.memory_requirements.memory_size,
+            )
+            return [
+                "Insufficient memory: %s < %s"
+                % (memory_size, profile.memory_requirements.memory_size)
+            ]
+
+        return []
+
+    def check_cluster_topology(self, profile: OpenSearchProfile) -> List[str]:
+        """Check the cluster topology requirements."""
+        cluster_fleet_apps = self.state.app.cluster_fleet_apps
+        current_app = self._current_peer_cluster_app()
+        # backwards compatibility for revisions that do not set generated roles
+        # in cluster_fleet_apps
+        if not cluster_fleet_apps or current_app.app.id in cluster_fleet_apps:
+            cluster_fleet_apps[current_app.app.id] = current_app
+
+        logger.debug("current_cluster_fleet_apps: %s", cluster_fleet_apps)
+        missing_requirements = []
+
+        nbr_cm_nodes = sum(
+            app.planned_units
+            for app in cluster_fleet_apps.values()
+            if "cluster_manager" in app.roles
+        )
+        nbr_data_nodes = sum(
+            app.planned_units for app in cluster_fleet_apps.values() if "data" in app.roles
+        )
+
+        if nbr_cm_nodes < profile.cluster_topology_requirements.cluster_managers:
+            missing_requirements.append(
+                f"At least {profile.cluster_topology_requirements.cluster_managers} cluster manager nodes are required. Found only {nbr_cm_nodes}."
```

Comment:
@Mehdi-Bendriss proposed to update this message to "At least: 3 cluster manager and 3 data nodes required." In the [How-To-Guide](https://docs.google.com/document/d/1yoGA-6_eVnR9U7QhpMQdakTNITnPj6J2W9Pbqiu40lE/edit?tab=t.0)